cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -40,24 +52,28 @@ func (c *CredentialsCmd) Run() error {
 	return err
 }
 
-var CLI struct {
+type CLI struct {
 	Credentials CredentialsCmd `kong:"cmd,help:'print credentials in a format usable as an AWS credentials_process'"`
 }
 
-func Run(args []string) error {
-	p, err := kong.New(&CLI)
+func Run(args []string, context *CliContext) error {
+	ctx, err := parse(args)
 	if err != nil {
 		return err
 	}
 
-	ctx, err := p.Parse(args)
-	if err != nil {
+	if err = ctx.Run(context); err != nil {
 		return err
 	}
 
-	if err = ctx.Run(); err != nil {
-		return err
+	return nil
+}
+
+func parse(args []string) (*kong.Context, error) {
+	parser, err := kong.New(&CLI{}, kong.NamedMapper(iso8601DurationMapperType, iso8601DurationMapper{}))
+	if err != nil {
+		return nil, err
 	}
 
-	return nil
+	return parser.Parse(args)
 }
```

Comment:
I extracted out the parsing logic into a separate function to facilitate unit testing. This way we can verify that a given command line will parse without also invoking the rest of the business logic (i.e. actually making a call to AWS).

Reply:
I like that!